### ISSUE TITLE
Navigating to different style text in MS outlook, NVDA often cannot find desired offset in TextInfo.

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#16406]
### Steps to reproduce:
1. Open NVDA
2. Assign [ and shift+[ for different style navigation 
3. Assign ] and shift+] for same style navigation
4. Open an Outlook email, see email attached, just remove the .zip characters at the end to open the email.
5. Make sure UIA for MS Word is disabled, this has also an effect on MS Outlook.
6. Press the assigned commands for navigating to next and previous text styles

### Actual behavior:
NVDA often gives a runtime error when attempting to navigate to next or previous different style text and is silent.
```
IO - inputCore.InputManager.executeGesture (22:50:08.993) - winInputHook (10508):
Input: kb(laptop):ü
ERROR - scriptHandler.executeScript (22:50:09.164) - MainThread (19244):
error executing script: <bound method BrowseModeTreeInterceptor.addQuickNav.<locals>.<lambda> of <appModules.outlook.MailViewerTreeInterceptor object at 0x0A1C4630>> with gesture 'ü'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 295, in executeScript
  File "browseMode.pyc", line 558, in <lambda>
  File "browseMode.pyc", line 516, in _quickNavScript
  File "browseMode.pyc", line 2447, in _iterTextStyle
  File "textInfos\__init__.pyc", line 793, in moveToCodepointOffset
RuntimeError: Unable to find desired offset in TextInfo.
```
[Webinar des Monats Förderung durch die öffentliche Hand.msg.zip](https://github.com/nvaccess/nvda/files/15002371/Webinar.des.Monats.Forderung.durch.die.offentliche.Hand.msg.zip)

### Expected behavior:
NVDA should report the next or the previous different style text properly,  or "no next different style text" and "no previous different style text", in case no such text style has been found. This works as expected for next and previous same style text commands.

### NVDA logs, crash dumps and other attachments:
n/a

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
Version: alpha-31612,3d4aOa8f (2024.2.0.31612)

#### Windows version:
Windows 11 23 H2

#### Name and version of other software in use when reproducing the issue:
Microsoft 365 MSO (Version 2403 Build 16.0.17425.20176) 64 Bit

#### Other information about your system:
n/a

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
n/a

#### If NVDA add-ons are disabled, is your problem still occurring?
yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Adriani90** [comment]
cc: @mltony since many users use Outlook without UIA, this needs fixed for the object model.

**Comment by mltony** [comment]
I'll try to fix it - sorry - I was down with cold for the last few days.

**Comment by mltony** [comment]
When UIA in Word is enabled, I fixed this in #16447.
However, when UIA in word is disabled, the reason for this exception seems to be completely different and I tracked this down to newly created #16448, which boils down to inconsistent text reporting by textInfo. I've never dealt with low-level textInfo implementations and not sure if I have enough cycles to fix the underlying textInfo issue.

**Comment by Adriani90** [comment]
#16447 only partially addresses this issue, that PR should not close this since many people still use MS Office without UIA because there are still alot of things which work better when it is disabled.
The full fix of this issue is probably blocked by #16448.
